src/components/Button/Base/Button.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import ButtonIcon from './ButtonIcon';
+
+const Button = (props) => {
+  const {
+    children,
+    className,
+    flavor,
+    href,
+    icon,
+    iconPosition,
+    sprite,
+    title,
+    ...rest
+  } = props;
+
+  const sldsClasses = cx(
+    'slds-button',
+    { [`slds-button_${flavor}`]: !!flavor },
+    className,
+  );
+
+  const ButtonEl = href ? 'a' : 'button';
+  const position = iconPosition || 'left';
+  const content = children || title;
+
+  const isShortcut = !!icon && !!sprite;
+  const isRightShortcut = isShortcut && position !== 'left';
+
+  return (
+    <ButtonEl
+      {...rest}
+      className={sldsClasses}
+      href={href}
+      title={title}
+    >
+      {isRightShortcut ? content : null}
+      {isShortcut && (
+        <ButtonIcon
+          position={position}
+          icon={icon}
+          sprite={sprite}
+        />
+      )}
+      {!isRightShortcut ? content : null}
+    </ButtonEl>
+  );
+};
+
+Button.defaultProps = {
+  children: null,
+  className: null,
+  flavor: 'neutral',
+  href: null,
+  icon: null,
+  iconPosition: null,
+  sprite: null,
+  title: null,
+};
+
+Button.propTypes = {
+  /**
+   * Used to set content. Can be a string, a `ButtonIcon` or a combination of those
```

Comment:
Technically it `can` be everything. I'd remove that addendum -- it implies specific child types.